src/mcp/tools/index.ts

```diff
@@ -3,11 +3,28 @@ import { ServerFeature } from "../types.js";
 import { authTools } from "./auth/index.js";
 import { dataconnectTools } from "./dataconnect/index.js";
 import { firestoreTools } from "./firestore/index.js";
-import { projectTools } from "./project/index.js";
+import { directoryTools } from "./directory/index.js";
+import { coreTools } from "./core/index.js";
 import { storageTools } from "./storage/index.js";
 
-export const tools: Record<ServerFeature, ServerTool[]> = {
-  project: projectTools,
+/** availableTools returns the list of MCP tools available given the server flags */
+export function availableTools(fixedRoot: boolean, activeFeatures?: ServerFeature[]): ServerTool[] {
+  // Core tools are always present.
+  const toolDefs: ServerTool[] = addPrefixToToolName("firebase_", coreTools);
+  if (!fixedRoot) {
+    // Present if the root is not fixed.
+    toolDefs.push(...directoryTools);
+  }
+  if (!activeFeatures || !activeFeatures.length) {
+    activeFeatures = Object.keys(tools) as ServerFeature[];
+  }
+  for (const key of activeFeatures) {
+    toolDefs.push(...tools[key]);
+  }
+  return toolDefs;
+}
```

Comment:
@mbleigh Let me now what you think.

I think this function makes the intention more clear.

The difference between directory tools and core tools were confusing.